test/TestUtilities/Test.Utility/Test.Utility.csproj

```diff
@@ -17,6 +17,7 @@
     <NoWarn>$(NoWarn);VSTHRD001;VSTHRD002;VSTHRD110</NoWarn>
     <Description>A collection of test utilities, such as generating packages, mocking servers, stub implementations of interfaces, etc.</Description>
     <SkipAnalyzers>true</SkipAnalyzers>
+    <NoPublicAPIs>true</NoPublicAPIs>
```

Comment:
I don't think this change should introduce a Directory.Build.Props file. 

We already achieve a similar thing through our common.props, we have props that are specific to projects, such as common.project.props. 

The trick is that the Test.Utility is probably built as a library rather than test code.

Reply:
I'd like to understand why test projects are being published as nupkgs. I feel like like neither workaround should be needed, but in order not to delay this getting in, this is sufficient to me.

Reply:
perhaps this should be "non public" apis for a project by default. And turn it on in common.props for non-test stuff?

---
In reply to: [458114772](https://github.com/NuGet/NuGet.Client/pull/3512#discussion_r458114772) [](ancestors = 458114772)

Reply:
So, the reason that Test.Utility is a nupkg is because the server team use it for creating certs for testing signed packages. I didn't check how often they update the version of the package they use. If they never do, they'll never notice if we stop packing this test project (and if they need an update, we could manually do it once-off, rather than on every build).

The second test project where I turned off the analyzer was a mistake. It was being packed, but I stopped packing it with this PR: https://github.com/NuGet/NuGet.Client/pull/3448   as a result, the MSBuild property about the public api analyzer is no longer needed (and I removed it).

Therefore, now there is only a single test project being packed as a nupkg. 

Also, I missed this from @dominoFire's first comment in this thread, not all projects in `src\` use per-tfm public APIs. In fact, I think there's only 5 that do, and I wish I could reduce that down to zero, but that's not possible for some projects, and problematic for others. I decided to make the "mininal" change here, adding the analyzers, but not modifying any production code in any way.